remote/remote_state_s3.go

```diff
@@ -560,10 +560,11 @@ func isBucketAlreadyOwnedByYouError(err error) bool {
 }
 
 // Add a policy to allow root access to the bucket
-func EnableRootAccesstoS3Bucket(s3Client *s3.S3, config *RemoteStateConfigS3, terragruntOptions *options.TerragruntOptions) error {
-	terragruntOptions.Logger.Debugf("Enabling root access to S3 bucket %s", config.Bucket)
+func EnableRootAccesstoS3Bucket(s3Client *s3.S3, config *ExtendedRemoteStateConfigS3, terragruntOptions *options.TerragruntOptions) error {
+	bucket := config.remoteStateConfigS3.Bucket
+	terragruntOptions.Logger.Debugf("Enabling root access to S3 bucket %s", bucket)
 
-	accountID, err := aws_helper.GetAWSAccountID(terragruntOptions)
+	accountID, err := aws_helper.GetAWSAccountID(config.GetAwsSessionConfig(), terragruntOptions)
```

Comment:
This is the source of the bug: this call was ignoring all the session parameters defined in the remote state config.